README.md

```diff
@@ -2,6 +2,32 @@
 
 [![Python 3.10](https://img.shields.io/badge/python-3.10-blue.svg)](https://www.python.org/downloads/release/python-3100)
 
+## Table of Contents
```

Comment:
Is there a precommit hook to generate this automatically? Worried it will get out of sync quickly. Also, I'm not sure how necessary this is since gh already generates a TOC here: 

![image](https://github.com/user-attachments/assets/2b809ebf-1683-4015-b9fc-2e1f4999e335)